src/accelerate/utils/modeling.py

```diff
@@ -771,8 +772,18 @@ def get_balanced_memory(
         )
 
     if num_devices == 1:
-        # We cannot do low_zero on just one GPU
+        # We cannot do low_zero on just one GPU, but we will still reserve some memory for the buffer
         low_zero = False
+        # If user just asked us to handle memory usage, we should avoid OOM
+        if user_not_set_max_memory:
+            for key in max_memory.keys():
+                if isinstance(key, int):
+                    max_memory[key] *= 0.9  # 90% is a good compromise
+                    logger.warning(
```

Comment:
Let's do an info here, as users are scared of warnings.

Reply:
done~